desc/compute/_geometry.py

```diff
@@ -234,154 +234,466 @@ def _a_major_over_a_minor(params, transforms, profiles, data, **kwargs):
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["n_theta", "e_zeta_z"],
+)
+def _L_sff_theta(params, transforms, profiles, data, **kwargs):
+    data["L_sff_theta"] = dot(data["e_zeta_z"], data["n_theta"])
+    return data
+
+
+@register_compute_fun(
+    name="M_sff_theta",
+    label="M_{\\mathrm{SFF},\\theta}",
+    units="m",
+    units_long="meters",
+    description="M coefficient of second fundamental form of constant theta surface",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["n_theta", "e_zeta_r"],
+)
+def _M_sff_theta(params, transforms, profiles, data, **kwargs):
+    data["M_sff_theta"] = dot(data["e_zeta_r"], data["n_theta"])
+    return data
+
+
+@register_compute_fun(
+    name="N_sff_theta",
+    label="N_{\\mathrm{SFF},\\theta}",
+    units="m",
+    units_long="meters",
+    description="N coefficient of second fundamental form of constant theta surface",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["n_theta", "e_rho_r"],
+)
+def _N_sff_theta(params, transforms, profiles, data, **kwargs):
+    data["N_sff_theta"] = dot(data["e_rho_r"], data["n_theta"])
+    return data
+
+
+@register_compute_fun(
+    name="curvature_k1_theta",
+    label="k_{1,\\theta}",
+    units="m^{-1}",
+    units_long="Inverse meters",
+    description="First principle curvature of constant theta surfaces",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["g_rr", "g_rz", "g_zz", "L_sff_theta", "M_sff_theta", "N_sff_theta"],
+)
+def _curvature_k1_theta(params, transforms, profiles, data, **kwargs):
+    # following notation from
+    # https://en.wikipedia.org/wiki/Parametric_surface#Curvature
+    E = data["g_rr"]
+    F = data["g_rz"]
+    G = data["g_zz"]
+    L = data["L_sff_theta"]
+    M = data["M_sff_theta"]
+    N = data["N_sff_theta"]
+    a = E * G - F**2
+    b = F * M - L * G - E * N
+    c = L * N - M**2
+    r1 = (-b + jnp.sqrt(b**2 - 4 * a * c)) / (2 * a)
+    r2 = (-b - jnp.sqrt(b**2 - 4 * a * c)) / (2 * a)
+    data["curvature_k1_theta"] = jnp.maximum(r1, r2)
+    data["curvature_k2_theta"] = jnp.minimum(r1, r2)
+    return data
+
+
+@register_compute_fun(
+    name="curvature_k2_theta",
+    label="k_{2,\\theta}",
+    units="m^{-1}",
+    units_long="Inverse meters",
+    description="Second principle curvature of constant theta surfaces",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["g_rr", "g_rz", "g_zz", "L_sff_theta", "M_sff_theta", "N_sff_theta"],
+)
+def _curvature_k2_theta(params, transforms, profiles, data, **kwargs):
+    # following notation from
+    # https://en.wikipedia.org/wiki/Parametric_surface#Curvature
+    E = data["g_rr"]
+    F = data["g_rz"]
+    G = data["g_zz"]
```

Comment:
Are you saying this because of the coordinate convention? The current coordinate system is (r, t, z) right now, meaning that we should flip E and G to maintain the same handedness.

Reply:
It was necessary to swap them so that `E = data["g_zz"]` and `G = data["g_rr"]` because $L$ in the second fundamental form should correspond to $E$ in the first fundamental form, likewise with $N$ and $G$. Since $L$ was the second derivative of the zeta coordinate vector $\partial_{\zeta} e_{\zeta}$ dotted with the normal, then E should be $g_{\zeta \zeta} = e_\zeta \cdot e_\zeta$